.gitignore

```diff
@@ -1,9 +1,29 @@
-/target
-
 # Test files
 /datacore
 /dataindex
 /dataldk0
 /dataldk1
 /dataldk2
 /tmp
+
+# will have compiled files and executables
+debug/
+/target
+
+# Remove Cargo.lock from gitignore if creating an executable, leave it for libraries
+# More information here https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html
+Cargo.lock
```

Comment:
`Cargo.lock` is included on purpose, so builds use specific dependency versions (e.g. when running tests), please drop this and restore the file itself

Reply:
done.